my-app/src/Shared/Subcategory/Subcategory.jsx

```diff
@@ -0,0 +1,27 @@
+import {NavDropdown} from "react-bootstrap";
+import useSubcategories from "./Services";
+import "../Header/Header.css"
+
+function Subcategory({ category, onSubcategoryClick }) {
+  const subcategories = useSubcategories(category.id)
+
+   const handleSubcategoryClick = (subcategory) => {
+    if (onSubcategoryClick) {
+      onSubcategoryClick(subcategory.title);
+    }
+  };
+
+  return (
+    <div>
+          <NavDropdown title={category?.title} id="categoryDropdown">
+            {subcategories && subcategories.map(subcategory => (
+              <NavDropdown.Item key={subcategory.id} href="#" onClick={() => handleSubcategoryClick(subcategory)}>
```

Comment:
we can query products against subcategory id as we are making it a foreign key in products, right?  we can only pass id in `handleSubcategoryClick` instead object.

Reply:
Yes we can do this, should I change it?

Reply:
If you can then let's do